ports/nrf/common-hal/neopixel_write/__init__.c

```diff
@@ -0,0 +1,161 @@
+    pinMask =  (1UL << (pin % 32));  // From port_pin_set_output_level ASF code.
+    ptr     =  pixels;
+    end     =  ptr + numBytes;
+    p       = *ptr++;
+    bitMask =  0x80;
+
+    volatile uint32_t *set = &(port->OUTSET.reg),
+                      *clr = &(port->OUTCLR.reg);
+
+    for(;;) {
+        *set = pinMask;
+        // This is the time where the line is always high regardless of the bit.
+        // For the SK6812 its 0.3us +- 0.15us
+        #ifdef SAMD21
+        asm("nop; nop;");
+        #endif
+        #ifdef SAMD51
+        delay_cycles(3);
+        #endif
+        if(p & bitMask) {
+            // This is the high delay unique to a one bit.
+            // For the SK6812 its 0.3us
+            #ifdef SAMD21
+            asm("nop; nop; nop; nop; nop; nop; nop;");
+            #endif
+            #ifdef SAMD51
+            delay_cycles(11);
+            #endif
+            *clr = pinMask;
+        } else {
+            *clr = pinMask;
+            // This is the low delay unique to a zero bit.
+            // For the SK6812 its 0.3us
+            #ifdef SAMD21
+            asm("nop; nop;");
+            #endif
+            #ifdef SAMD51
+            delay_cycles(3);
+            #endif
+        }
+        if((bitMask >>= 1) != 0) {
+            // This is the delay between bits in a byte and is the 1 code low
+            // level time from the datasheet.
+            // For the SK6812 its 0.6us +- 0.15us
+            #ifdef SAMD21
+            asm("nop; nop; nop; nop; nop;");
+            #endif
+            #ifdef SAMD51
+            delay_cycles(20);
+            #endif
+        } else {
+            if(ptr >= end) break;
+            p       = *ptr++;
+            bitMask = 0x80;
+            // This is the delay between bytes. It's similar to the other branch
+            // in the if statement except its tuned to account for the time the
+            // above operations take.
+            // For the SK6812 its 0.6us +- 0.15us
+            #ifdef SAMD51
+            delay_cycles(15);
+            #endif
+        }
+    }
+
+    #ifdef SAMD21
+    // Speed up! (But inconsistent timing.)
+    NVMCTRL->CTRLB.bit.READMODE = NVMCTRL_CTRLB_READMODE_NO_MISS_PENALTY_Val;
+    #endif
+
+    #ifdef SAMD51
+    // Turn instruction, data, and NVM caches back on.
+    hri_cmcc_clear_CFG_reg(CMCC, CMCC_CFG_DCDIS | CMCC_CFG_ICDIS);
+    hri_nvmctrl_clear_CTRLA_CACHEDIS0_bit(NVMCTRL);
+    hri_nvmctrl_clear_CTRLA_CACHEDIS1_bit(NVMCTRL);
+
+    #endif
+
+    // ticks_ms may be out of date at this point because we stopped the
+    // interrupt. We'll risk it anyway.
+    current_tick(&next_start_tick_ms, &next_start_tick_us);
+    if (next_start_tick_us < 100) {
+        next_start_tick_ms += 1;
+        next_start_tick_us = 100 - next_start_tick_us;
+    } else {
+        next_start_tick_us -= 100;
+    }
+
+    // Turn on interrupts after timing-sensitive code.
+    mp_hal_enable_all_interrupts();
+*/
```

Comment:
No need to check in all of this commented out code. It's still available in the atmel port.

Reply:
removed commented-out code, and it's about to replaced by P$ #1161.